src/compton/Compton.hh

```diff
@@ -68,13 +68,22 @@ public:
   explicit Compton(const std::string &filehandle);
 
   //! Constructor to build a multigroup library from an existing pointwise file
+  //! Defaulting the number of angular evals (n_xi) to zero causes the CSK
+  //! routines to use the full angular fidelity of the library
   Compton(const std::string &file, const std::vector<double> &group_bounds,
-          const size_t n_xi);
-
-  //! Interpolation of all data to a certain electron temperature:
-  std::vector<std::vector<std::vector<double>>> interpolate(const double etemp);
-  std::vector<std::vector<std::vector<double>>>
-  interpolate(const double etemp) const;
+          const std::string opac_type, const std::string wt_func,
+          const bool induced, const size_t n_xi = 0);
+
+  //! Interpolation of all csk data to a certain electron temperature:
+  std::vector<std::vector<std::vector<std::vector<double>>>>
+  interpolate_csk(const double etemp);
+  std::vector<std::vector<std::vector<std::vector<double>>>>
+  interpolate_csk(const double etemp) const;
```

Comment:
I honestly forgot that you could call const member functions on non-const objects! I'm going to ditch all the non-const calls, since none of the functions are modifying state...